Activate release 11.2.1/Add 429 error mitigations details (#207)
giantswarm/releases: release-notes/azure/v11.2.1.md

## :zap: Giant Swarm Release 11.2.1 for Azure :zap:

### This release contains Kubernetes ```1.16.8``` and other changes that increase cluster resiliency against API Throttling (error 429). Additional improvements include:
- **Failed provisioning of VMs does not cause 429 anymore.**  We have added an automation that will reimage the failed instance immediately, before it exhausts API rate limits.

> - **Freeze of upgrade if API rate limits are almost exhausted.** Upgrades are causing a movement in deployments, updating templates as well as nodes, triggering more calls to API. Upgraded will be freezed automatically when limits are almost reached and resumed after situation is back to normal.

Reviewer: ```suggestion
- **Freeze of upgrade if API rate limits are almost exhausted.** Upgrades are causing a movement in deployments, updating templates as well as nodes, triggering more calls to the API. Upgraded resources will be frozen automatically when limits are almost reached, and resumed after situation is back to normal.
```

Author: I can see I worded it incorrectly, but it is about the whole upgrade process that will be frozen, can I add resources there?

Reviewer: I'm not 100% sure about the `Upgraded resources` part, I saw that there was something missing, but I  was hoping you'd know what is actually missing 😬

Reviewer: > @axbarsan (didn't know how to do it better so I edited your suggestion, sorry!)
> 
> > Upgrade process will be frozen automatically
> 
> Is that fine?

Yes, that's perfect 👍

Author: Perfect, thank you very much for your suggestions!! 😁 

freezed - this will now haunt me 🤦‍♀

> - **Respecting the ```Retry After``` header returned by Azure API,** because not doing so made clusters facing the 429 error get stuck, together with every other cluster in the same subscription. Now clusters are able to recover by themselves and do not affect other clusters in the subscription.

Reviewer 2: ```suggestion
- **Respecting the ```Retry After``` header returned by Azure API,** because not doing so made clusters face the 429 error and get stuck, together with every other cluster in the same subscription. Now clusters are able to recover by themselves and do not affect other clusters in the subscription.
```

Author: Oh my understanding was that clusters still can face 429 but won't get stuck on it, they will hit it once and wait before doing anything again, that's why I worded it like this.

> - **Respecting the ```Retry After``` header returned by Azure API** Because it was not respected before clusters that faced 429 got stuck, together with every other cluster in the same subscription. Now clusters are able to recover by themselves and do not affect other clusters in the subscription.

Reviewer: ```suggestion
- **Respecting the ```Retry After``` header returned by Azure API,** because not doing so made clusters facing the 429 error get stuck, together with every other cluster in the same subscription. Now clusters are able to recover by themselves and do not affect other clusters in the subscription.
```
